teambit.bvm/link/link.ts

```diff
@@ -105,11 +114,46 @@ function getBitBinPath(){
   return path.join('@teambit', 'bit', 'bin', 'bit');
 }
 
-function validateBinDirInPath(binDir: string){
+async function validateBinDirInPath(binDir: string, opts: { updatePath?: boolean } = { updatePath: true }) {
   const osPaths = (process.env.PATH || process.env.Path || process.env.path).split(path.delimiter);
-  if (osPaths.indexOf(binDir) === -1) {
+  if (osPaths.indexOf(binDir) !== -1) return;
+  if (!opts.updatePath) {
     const err = IS_WINDOWS ? windowsMissingInPathError(binDir, WINDOWS_INSTALL_TROUBLESHOOTING_DOCS_URL) : macLinuxMissingInPathError(binDir, MAC_LINUX_INSTALL_TROUBLESHOOTING_DOCS_URL);
     console.log(chalk.yellowBright(err));
+  } else {
+    const report = await addDirToEnvPath(binDir, {
+      overwrite: true,
+      position: 'end',
+      configSectionName: 'bit',
+    })
+    const output = renderSetupOutput(report)
+    if (output) {
+      console.log(output)
+    }
+  }
+}
+
+function renderSetupOutput (report: PathExtenderReport): string | undefined {
+  if (report.oldSettings === report.newSettings) {
+    return undefined
+  }
+  const output = []
+  if (report.configFile) {
+    output.push(reportConfigChange(report.configFile))
+  }
+  output.push(`Next configuration changes were made:
+${report.newSettings}
+${chalk.blueBright('Setup complete. Open a new terminal to start using Bit.')}
+`)
+  return output.join('\n')
+}
+
+function reportConfigChange (configReport: ConfigReport): string {
+  switch (configReport.changeType) {
+  case 'created': return `Created ${configReport.path}`
+  case 'appended': return `Appended new lines to ${configReport.path}`
+  case 'modified': return `Replaced configuration in ${configReport.path}`
+  case 'skipped': return `Configuration already up-to-date in ${configReport.path}`
```

Comment:
move to the command layer

Reply:
moved to a separate component that is used from the command layer